### PR TITLE
fix: update context must emit events

### PR DESCRIPTION
**Pull request by FrazerBayley**

### CODE REVIEW

File: src/tests/context.test.js [hunk 1]
```diff
@@ -72,5 +73,58 @@ describe('context', () => {
...
+      context.update(
+        {
+          test: {
+            boolean: true
+          }
+        }
+      )
+    });
+
+    it('should emit CONTEXT_VALUE_CHANGED when key is added to context', (done) => {
```
Comment (mattstrom) [inline src/tests/context.test.js:113]: Copy-pasta mistake?
```suggestion
    it('should emit CONTEXT_VALUE_CHANGED when key is updated in context', (done) => {
```

Comment (FrazerBayley) [reply src/tests/context.test.js:113]: Yep

File: src/context.js [hunk 2]
```diff
@@ -145,12 +145,13 @@ function EvolvContext(store) {
       context = remoteContext;
     }
 
+    const thisRef = this;
```
Comment (mattstrom) [inline src/context.js:148]: Not wrong, but I don't think capturing `this` as another variable was necessary.

Comment (FrazerBayley) [reply src/context.js:148]: It was for some reason. Inside the forEach `this` was undefined. Idk why, do you?

**Pull request merged**